### PR TITLE
gnrc_lorawan: Ensure minimal packet length

### PR DESCRIPTION
**Pull request by Teufelchen1**
### Contribution description

Hello 🦛

This adds missing bounds checks.

### Testing procedure

Honestly, I am just hoping for CI 🫤

### CODE REVIEW

File: sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c [hunk 1]
```diff
@@ -116,6 +116,10 @@ int gnrc_lorawan_parse_dl(gnrc_lorawan_t *mac, uint8_t *buf, size_t len,
         pkt->fopts.iol_base = buf;
         pkt->fopts.iol_len = fopts_length;
         buf += fopts_length;
+        if (buf >= p_mic) {
```
Comment (benpicco) [inline sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c:119]: is this not already covered by the `if (buf < p_mic)`check below?

Comment (Teufelchen1) [reply sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c:119]: I feel stupid as I am not seeing your argument? Could you elaborate?

Comment (benpicco) [reply sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c:119]: `buf >= p_mic` is the same as `buf < p_mic`, there is also a check for `fopts_length` in there (but only when port is 0?), so this code path is already (almost?) covered.

Comment (benpicco) [reply sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c:119]: Sorry got confused myself, I there was an `else` branch to the `if (buf < p_mic)` I guess `buf >= p_mic` is only a problem in the `fopts_length` case then? (arg, this code could need some comments)

Comment (benpicco) [reply sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c:119]: None at all

**Review by benpicco — APPROVED** [review]

**Pull request merged**